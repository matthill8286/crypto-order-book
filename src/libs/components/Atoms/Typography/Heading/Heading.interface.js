
//# sourceMappingURL=Heading.interface.js.map