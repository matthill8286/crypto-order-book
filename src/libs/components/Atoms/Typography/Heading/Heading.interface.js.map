{"version":3,"sources":[],"names":[],"mappings":"","sourcesContent":[],"file":"Heading.interface.js"}