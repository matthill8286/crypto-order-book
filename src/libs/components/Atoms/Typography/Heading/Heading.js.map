{"version":3,"sources":["../../../../../src/components/Atoms/Typography/Heading/Heading.tsx"],"names":["React","Typo","HeadingPresets","Heading","scale","margin","children","bold","uppercase","limitLines","limitWidth","color","showCursor","restProps","props","fontFamily","spacing","weight","displayName"],"mappings":";;;;;;;;;;;;AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,IAAT;AAGA,SAASC,cAAT;AAEA,OAAO,IAAMC,OAA+B,GAAG,SAAlCA,OAAkC,OAWzC;AAAA,MAVJC,KAUI,QAVJA,KAUI;AAAA,MATJC,MASI,QATJA,MASI;AAAA,MARJC,QAQI,QARJA,QAQI;AAAA,uBAPJC,IAOI;AAAA,MAPJA,IAOI,0BAPG,KAOH;AAAA,4BANJC,SAMI;AAAA,MANJA,SAMI,+BANQ,KAMR;AAAA,MALJC,UAKI,QALJA,UAKI;AAAA,MAJJC,UAII,QAJJA,UAII;AAAA,MAHJC,KAGI,QAHJA,KAGI;AAAA,MAFJC,UAEI,QAFJA,UAEI;AAAA,MADDC,SACC;;AACJ,MAAMC,KAAgB;AACpBT,IAAAA,MAAM,EAAE,GADY;AAEpBM,IAAAA,KAAK,EAAEA,KAFa;AAGpBI,IAAAA,UAAU,EAAE,SAHQ;AAIpBC,IAAAA,OAAO,EAAE;AAJW,KAKjBd,cAAc,CAACE,KAAD,CALG,GAMjBS,SANiB,CAAtB;;AASA,MAAIR,MAAJ,EAAY;AACVS,IAAAA,KAAK,CAACT,MAAN,GAAeA,MAAf;AACD;;AAED,MAAIE,IAAJ,EAAU;AACRO,IAAAA,KAAK,CAACG,MAAN,GAAe,MAAf;AACD;;AAED,SACE,oBAAC,IAAD;AAAM,IAAA,UAAU,EAAER,UAAlB;AAA8B,IAAA,UAAU,EAAEC,UAA1C;AAAsD,IAAA,WAAW,EAAEF;AAAnE,KAAkFM,KAAlF,GACGR,QADH,CADF;AAKD,CAlCM;AAoCPH,OAAO,CAACe,WAAR,GAAsB,SAAtB","sourcesContent":["import React from 'react'\nimport { Typo } from '../Typo'\nimport { TypoProps } from '../Typo/Typo.interface'\nimport { HeadingProps } from './Heading.interface'\nimport { HeadingPresets } from './HeadingPresets'\n\nexport const Heading: React.FC<HeadingProps> = ({\n  scale,\n  margin,\n  children,\n  bold = false,\n  uppercase = false,\n  limitLines,\n  limitWidth,\n  color,\n  showCursor,\n  ...restProps\n}) => {\n  const props: TypoProps = {\n    margin: '0',\n    color: color,\n    fontFamily: 'default',\n    spacing: 'base',\n    ...HeadingPresets[scale],\n    ...restProps,\n  }\n\n  if (margin) {\n    props.margin = margin\n  }\n\n  if (bold) {\n    props.weight = 'bold'\n  }\n\n  return (\n    <Typo limitLines={limitLines} limitWidth={limitWidth} toUpperCase={uppercase} {...props}>\n      {children}\n    </Typo>\n  )\n}\n\nHeading.displayName = 'Heading'\n"],"file":"Heading.js"}