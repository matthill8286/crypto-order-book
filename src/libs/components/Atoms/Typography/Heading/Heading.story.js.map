{"version":3,"sources":["../../../../../src/components/Atoms/Typography/Heading/Heading.story.tsx"],"names":["boolean","select","text","storiesOf","React","saiyanTheme","Heading","HeadingFeatured","stories","module","knobs","scale","tag","bold","color","margin","fontFamily","showCursor","uppercase","Object","keys","font","family","add","sampleText","fixedSize","undefined","fixedMdSize"],"mappings":"AAAA,SAASA,OAAT,EAAkBC,MAAlB,EAA0BC,IAA1B,QAAsC,wBAAtC;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,WAAT;AAEA,SAASC,OAAT;AAEA,SAASC,eAAT;AAEA,IAAMC,OAAO,GAAGL,SAAS,CAAC,gCAAD,EAAmCM,MAAnC,CAAzB;;AAEA,IAAMC,KAAK,GAAG,SAARA,KAAQ;AAAA,wBACZC,KADY;AAAA,MACZA,KADY,2BACJ,SADI;AAAA,sBAEZC,GAFY;AAAA,MAEZA,GAFY,yBAEN,IAFM;AAAA,uBAGZC,IAHY;AAAA,MAGZA,IAHY,0BAGL,KAHK;AAAA,wBAIZC,KAJY;AAAA,MAIZA,KAJY,2BAIJ,OAJI;AAAA,yBAKZC,MALY;AAAA,MAKZA,MALY,4BAKH,EALG;AAAA,6BAMZC,UANY;AAAA,MAMZA,UANY,gCAMC,SAND;AAAA,6BAOZC,UAPY;AAAA,MAOZA,UAPY,gCAOC,KAPD;AAAA,SAQP;AACLN,IAAAA,KAAK,EAAEV,MAAM,CACX,YADW,EAEX,CAAC,SAAD,EAAY,SAAZ,EAAuB,SAAvB,EAAkC,SAAlC,EAA6C,SAA7C,CAFW,EAGXU,KAHW,CADR;AAMLC,IAAAA,GAAG,EAAEX,MAAM,CACT,UADS,EAET,CAAC,IAAD,EAAO,IAAP,EAAa,IAAb,EAAmB,IAAnB,EAAyB,IAAzB,EAA+B,QAA/B,EAAyC,MAAzC,CAFS,EAGTW,GAHS,CANN;AAWLC,IAAAA,IAAI,EAAEb,OAAO,CAAC,MAAD,EAASa,IAAT,CAXR;AAYLK,IAAAA,SAAS,EAAElB,OAAO,CAAC,WAAD,EAAc,KAAd,CAZb;AAaLiB,IAAAA,UAAU,EAAEjB,OAAO,CAAC,aAAD,EAAgB,KAAhB,CAbd;AAcLc,IAAAA,KAAK,EAAEb,MAAM,CAAC,OAAD,EAAUkB,MAAM,CAACC,IAAP,CAAYf,WAAW,CAACS,KAAxB,CAAV,EAA0CA,KAA1C,CAdR;AAeLC,IAAAA,MAAM,EAAEd,MAAM,CAAC,QAAD,EAAW,CAAC,EAAD,EAAK,IAAL,EAAW,MAAX,EAAmB,cAAnB,EAAmC,GAAnC,CAAX,EAAoDc,MAApD,CAfT;AAgBLC,IAAAA,UAAU,EAAEf,MAAM,CAChB,YADgB,EAEhBkB,MAAM,CAACC,IAAP,CAAYf,WAAW,CAACgB,IAAZ,CAAiBC,MAA7B,CAFgB,EAGhBN,UAHgB;AAhBb,GARO;AAAA,CAAd;;AA+BAR,OAAO,CAACe,GAAR,CAAY,SAAZ,EAAuB,YAAM;AAC3B,MAAMC,UAAU,GAAGtB,IAAI,CAAC,cAAD,sBAAvB;AAEA,SAAO,oBAAC,OAAD,EAAaQ,KAAK,CAAC,EAAD,CAAlB,EAAyBc,UAAzB,CAAP;AACD,CAJD;AAMAhB,OAAO,CAACe,GAAR,CAAY,kBAAZ,EAAgC,YAAM;AACpC,MAAMC,UAAU,GAAGtB,IAAI,CAAC,cAAD,8BAAvB;AACA,MAAMuB,SAAS,GAAGxB,MAAM,CAAC,YAAD,EAAe,CAAC,EAAD,EAAK,IAAL,EAAW,MAAX,CAAf,EAAmCyB,SAAnC,CAAxB;AACA,MAAMC,WAAW,GAAG3B,OAAO,CAAC,wBAAD,EAA2B,KAA3B,CAA3B;AACA,SACE,oBAAC,eAAD;AAAiB,IAAA,GAAG,EAAC,IAArB;AAA0B,IAAA,SAAS,EAAEyB,SAArC;AAAgD,IAAA,WAAW,EAAEE,WAA7D;AAA0E,IAAA,KAAK,EAAC;AAAhF,KACGH,UADH,CADF;AAKD,CATD","sourcesContent":["import { boolean, select, text } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { saiyanTheme } from '@/styles/sc-vars-saiyan'\nimport { ThemeColors, ThemeFontFamilies } from '@/types/theme'\nimport { Heading } from './Heading'\nimport { TypographyScaleHeadline } from './Heading.interface'\nimport { HeadingFeatured } from './HeadingFeatured'\n\nconst stories = storiesOf('Design System/Atoms/Typography', module)\n\nconst knobs = ({\n  scale = 'level-1',\n  tag = 'h1',\n  bold = false,\n  color = 'grey6',\n  margin = '',\n  fontFamily = 'default',\n  showCursor = false,\n}) => ({\n  scale: select(\n    'Type scale',\n    ['level-1', 'level-2', 'level-3', 'level-4', 'level-5'],\n    scale\n  ) as TypographyScaleHeadline,\n  tag: select(\n    'Tag name',\n    ['h1', 'h2', 'h3', 'h4', 'h5', 'strong', 'span'],\n    tag\n  ) as keyof JSX.IntrinsicElements,\n  bold: boolean('Bold', bold),\n  uppercase: boolean('uppercase', false),\n  showCursor: boolean('Show Cursor', false),\n  color: select('Color', Object.keys(saiyanTheme.color), color) as ThemeColors,\n  margin: select('margin', ['', 'xl', 'xl 0', 'md xl xxl xl', '0'], margin),\n  fontFamily: select(\n    'fontFamily',\n    Object.keys(saiyanTheme.font.family),\n    fontFamily\n  ) as ThemeFontFamilies,\n})\n\nstories.add('Heading', () => {\n  const sampleText = text('Example text', `Lorem ipsum dolor`)\n\n  return <Heading {...knobs({})}>{sampleText}</Heading>\n})\n\nstories.add('Heading Featured', () => {\n  const sampleText = text('Example text', `Featured Text for Teasers`)\n  const fixedSize = select('fixed size', ['', 'xl', 'xxxl'], undefined)\n  const fixedMdSize = boolean('medium as maximum size', false)\n  return (\n    <HeadingFeatured tag=\"h2\" fixedSize={fixedSize} fixedMdSize={fixedMdSize} color=\"secondary\">\n      {sampleText}\n    </HeadingFeatured>\n  )\n})\n"],"file":"Heading.story.js"}