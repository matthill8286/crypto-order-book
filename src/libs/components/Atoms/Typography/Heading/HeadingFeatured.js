function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { css, media, styled } from "../../../../styles";
import { getColor } from "../../../../utils/helper";
import { Typo } from "../Typo";
var StyledFeaturedHeading = styled(Typo).withConfig({
  displayName: "HeadingFeatured__StyledFeaturedHeading",
  componentId: "sc-1i1kuqv-0"
})(function (_ref) {
  var theme = _ref.theme,
      fixedSize = _ref.fixedSize,
      fixedMdSize = _ref.fixedMdSize;
  return css(["font-weight:bold;letter-spacing:normal;max-height:fit-content;padding-top:", ";text-transform:", ";line-height:", ";font-size:", ";", "{max-height:100%;}", ""], theme.heading.featured.fontPadding, theme.heading.featured.textTransform, theme.heading.featured.lineHeight.xs, fixedSize ? theme.font.size[fixedSize] : theme.heading.featured.fontSize, media.ie11, !fixedSize && css(["", "{font-size:", ";line-height:", ";}", ""], media.md, theme.font.size.xxxxl, theme.heading.featured.lineHeight.md, !fixedMdSize && css(["", "{font-size:", ";line-height:", ";}"], media.xl, theme.heading.featured.fontSize, theme.heading.featured.lineHeight.xl)));
});
export var HeadingFeatured = function HeadingFeatured(_ref2) {
  var children = _ref2.children,
      _ref2$uppercase = _ref2.uppercase,
      uppercase = _ref2$uppercase === void 0 ? false : _ref2$uppercase,
      _ref2$limitLines = _ref2.limitLines,
      limitLines = _ref2$limitLines === void 0 ? 2 : _ref2$limitLines,
      _ref2$color = _ref2.color,
      color = _ref2$color === void 0 ? 'primary' : _ref2$color,
      _ref2$tag = _ref2.tag,
      tag = _ref2$tag === void 0 ? 'p' : _ref2$tag,
      restProps = _objectWithoutProperties(_ref2, ["children", "uppercase", "limitLines", "color", "tag"]);

  var props = _objectSpread({
    margin: '0',
    tag: tag,
    color: getColor({
      defaultColor: 'grey6',
      color: color
    }),
    fontFamily: 'featured',
    fontSize: 'xxxxl',
    spacing: 'base'
  }, restProps);

  return React.createElement(StyledFeaturedHeading, _extends({
    limitLines: limitLines,
    toUpperCase: uppercase
  }, props), children);
};
HeadingFeatured.displayName = 'HeadingFeatured';
//# sourceMappingURL=HeadingFeatured.js.map