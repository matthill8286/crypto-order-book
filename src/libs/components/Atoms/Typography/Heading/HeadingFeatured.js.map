{"version":3,"sources":["../../../../../src/components/Atoms/Typography/Heading/HeadingFeatured.tsx"],"names":["React","css","media","styled","getColor","Typo","StyledFeaturedHeading","theme","fixedSize","fixedMdSize","heading","featured","fontPadding","textTransform","lineHeight","xs","font","size","fontSize","ie11","md","xxxxl","xl","HeadingFeatured","children","uppercase","limitLines","color","tag","restProps","props","margin","defaultColor","fontFamily","spacing","displayName"],"mappings":";;;;;;;;;;;;AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,GAAT,EAAcC,KAAd,EAAqBC,MAArB;AACA,SAASC,QAAT;AACA,SAASC,IAAT;AAIA,IAAMC,qBAAqB,GAAGH,MAAM,CAACE,IAAD,CAAT;AAAA;AAAA;AAAA,GACzB;AAAA,MAAGE,KAAH,QAAGA,KAAH;AAAA,MAAUC,SAAV,QAAUA,SAAV;AAAA,MAAqBC,WAArB,QAAqBA,WAArB;AAAA,SAAuCR,GAAvC,oKAIiBM,KAAK,CAACG,OAAN,CAAcC,QAAd,CAAuBC,WAJxC,EAKoBL,KAAK,CAACG,OAAN,CAAcC,QAAd,CAAuBE,aAL3C,EAMiBN,KAAK,CAACG,OAAN,CAAcC,QAAd,CAAuBG,UAAvB,CAAkCC,EANnD,EAOeP,SAAS,GAAGD,KAAK,CAACS,IAAN,CAAWC,IAAX,CAAgBT,SAAhB,CAAH,GAAgCD,KAAK,CAACG,OAAN,CAAcC,QAAd,CAAuBO,QAP/E,EAQIhB,KAAK,CAACiB,IARV,EAWI,CAACX,SAAD,IACAP,GADA,iDAEIC,KAAK,CAACkB,EAFV,EAGiBb,KAAK,CAACS,IAAN,CAAWC,IAAX,CAAgBI,KAHjC,EAImBd,KAAK,CAACG,OAAN,CAAcC,QAAd,CAAuBG,UAAvB,CAAkCM,EAJrD,EAOI,CAACX,WAAD,IACAR,GADA,6CAEIC,KAAK,CAACoB,EAFV,EAGiBf,KAAK,CAACG,OAAN,CAAcC,QAAd,CAAuBO,QAHxC,EAImBX,KAAK,CAACG,OAAN,CAAcC,QAAd,CAAuBG,UAAvB,CAAkCQ,EAJrD,CAPJ,CAXJ;AAAA,CADyB,CAA3B;AA8BA,OAAO,IAAMC,eAA+C,GAAG,SAAlDA,eAAkD,QAOzD;AAAA,MANJC,QAMI,SANJA,QAMI;AAAA,8BALJC,SAKI;AAAA,MALJA,SAKI,gCALQ,KAKR;AAAA,+BAJJC,UAII;AAAA,MAJJA,UAII,iCAJS,CAIT;AAAA,0BAHJC,KAGI;AAAA,MAHJA,KAGI,4BAHI,SAGJ;AAAA,wBAFJC,GAEI;AAAA,MAFJA,GAEI,0BAFE,GAEF;AAAA,MADDC,SACC;;AACJ,MAAMC,KAAgB;AACpBC,IAAAA,MAAM,EAAE,GADY;AAEpBH,IAAAA,GAAG,EAAEA,GAFe;AAGpBD,IAAAA,KAAK,EAAEvB,QAAQ,CAAC;AACd4B,MAAAA,YAAY,EAAE,OADA;AAEdL,MAAAA,KAAK,EAALA;AAFc,KAAD,CAHK;AAOpBM,IAAAA,UAAU,EAAE,UAPQ;AAQpBf,IAAAA,QAAQ,EAAE,OARU;AASpBgB,IAAAA,OAAO,EAAE;AATW,KAUjBL,SAViB,CAAtB;;AAaA,SACE,oBAAC,qBAAD;AAAuB,IAAA,UAAU,EAAEH,UAAnC;AAA+C,IAAA,WAAW,EAAED;AAA5D,KAA2EK,KAA3E,GACGN,QADH,CADF;AAKD,CA1BM;AA4BPD,eAAe,CAACY,WAAhB,GAA8B,iBAA9B","sourcesContent":["import React from 'react'\nimport { css, media, styled } from '@/styles'\nimport { getColor } from '@/utils/helper'\nimport { Typo } from '../Typo'\nimport { TypoProps } from '../Typo/Typo.interface'\nimport { HeadingFeaturedProps, StyledHeadingFeaturedProps } from './Heading.interface'\n\nconst StyledFeaturedHeading = styled(Typo)<StyledHeadingFeaturedProps>(\n  ({ theme, fixedSize, fixedMdSize }) => css`\n    font-weight: bold;\n    letter-spacing: normal;\n    max-height: fit-content;\n    padding-top: ${theme.heading.featured.fontPadding};\n    text-transform: ${theme.heading.featured.textTransform};\n    line-height: ${theme.heading.featured.lineHeight.xs};\n    font-size: ${fixedSize ? theme.font.size[fixedSize] : theme.heading.featured.fontSize};\n    ${media.ie11} {\n      max-height: 100%;\n    }\n    ${!fixedSize &&\n      css`\n        ${media.md} {\n          font-size: ${theme.font.size.xxxxl};\n          line-height: ${theme.heading.featured.lineHeight.md};\n        }\n\n        ${!fixedMdSize &&\n          css`\n            ${media.xl} {\n              font-size: ${theme.heading.featured.fontSize};\n              line-height: ${theme.heading.featured.lineHeight.xl};\n            }\n          `}\n      `}\n  `\n)\n\nexport const HeadingFeatured: React.FC<HeadingFeaturedProps> = ({\n  children,\n  uppercase = false,\n  limitLines = 2,\n  color = 'primary',\n  tag = 'p',\n  ...restProps\n}) => {\n  const props: TypoProps = {\n    margin: '0',\n    tag: tag,\n    color: getColor({\n      defaultColor: 'grey6',\n      color,\n    }),\n    fontFamily: 'featured',\n    fontSize: 'xxxxl',\n    spacing: 'base',\n    ...restProps,\n  }\n\n  return (\n    <StyledFeaturedHeading limitLines={limitLines} toUpperCase={uppercase} {...props}>\n      {children}\n    </StyledFeaturedHeading>\n  )\n}\n\nHeadingFeatured.displayName = 'HeadingFeatured'\n"],"file":"HeadingFeatured.js"}