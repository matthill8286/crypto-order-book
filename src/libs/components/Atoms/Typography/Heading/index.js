export * from "./Heading";
export * from "./HeadingFeatured";
export * from "./Heading.interface";
//# sourceMappingURL=index.js.map