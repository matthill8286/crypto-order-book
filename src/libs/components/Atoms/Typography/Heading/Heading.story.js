import { boolean, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { saiyanTheme } from "../../../../styles/sc-vars-saiyan";
import { Heading } from "./Heading";
import { HeadingFeatured } from "./HeadingFeatured";
var stories = storiesOf('Design System/Atoms/Typography', module);

var knobs = function knobs(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'level-1' : _ref$scale,
      _ref$tag = _ref.tag,
      tag = _ref$tag === void 0 ? 'h1' : _ref$tag,
      _ref$bold = _ref.bold,
      bold = _ref$bold === void 0 ? false : _ref$bold,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'grey6' : _ref$color,
      _ref$margin = _ref.margin,
      margin = _ref$margin === void 0 ? '' : _ref$margin,
      _ref$fontFamily = _ref.fontFamily,
      fontFamily = _ref$fontFamily === void 0 ? 'default' : _ref$fontFamily,
      _ref$showCursor = _ref.showCursor,
      showCursor = _ref$showCursor === void 0 ? false : _ref$showCursor;
  return {
    scale: select('Type scale', ['level-1', 'level-2', 'level-3', 'level-4', 'level-5'], scale),
    tag: select('Tag name', ['h1', 'h2', 'h3', 'h4', 'h5', 'strong', 'span'], tag),
    bold: boolean('Bold', bold),
    uppercase: boolean('uppercase', false),
    showCursor: boolean('Show Cursor', false),
    color: select('Color', Object.keys(saiyanTheme.color), color),
    margin: select('margin', ['', 'xl', 'xl 0', 'md xl xxl xl', '0'], margin),
    fontFamily: select('fontFamily', Object.keys(saiyanTheme.font.family), fontFamily)
  };
};

stories.add('Heading', function () {
  var sampleText = text('Example text', "Lorem ipsum dolor");
  return React.createElement(Heading, knobs({}), sampleText);
});
stories.add('Heading Featured', function () {
  var sampleText = text('Example text', "Featured Text for Teasers");
  var fixedSize = select('fixed size', ['', 'xl', 'xxxl'], undefined);
  var fixedMdSize = boolean('medium as maximum size', false);
  return React.createElement(HeadingFeatured, {
    tag: "h2",
    fixedSize: fixedSize,
    fixedMdSize: fixedMdSize,
    color: "secondary"
  }, sampleText);
});
//# sourceMappingURL=Heading.story.js.map