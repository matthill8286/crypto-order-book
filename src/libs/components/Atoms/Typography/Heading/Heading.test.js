import React from 'react';
import { renderWithTheme } from "../../../../testRenderer";
import { Heading } from "./Heading";
describe('Heading', function () {
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(Heading, {
      scale: "level-1",
      fontFamily: "featured"
    }));
    expect(tree).toMatchSnapshot();
  });
  var scaleOptions = ['level-1', 'level-2', 'level-3', 'level-4', 'level-5'];
  scaleOptions.forEach(function (scale) {
    it("should match snapshot with scale=\"".concat(scale, "\""), function () {
      var tree = renderWithTheme(React.createElement(Heading, {
        scale: scale
      }));
      expect(tree).toMatchSnapshot();
    });
  });
});
//# sourceMappingURL=Heading.test.js.map