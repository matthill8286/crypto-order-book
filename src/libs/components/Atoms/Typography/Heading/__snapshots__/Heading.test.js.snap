// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly 1`] = `
<h1
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 ctdUyT"
  fontFamily="featured"
  fontSize="xxxl"
  spacing="base"
/>
`;

exports[`Heading should match snapshot with scale="level-1" 1`] = `
<h1
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 draCiN"
  fontFamily="default"
  fontSize="xxxl"
  spacing="base"
/>
`;

exports[`Heading should match snapshot with scale="level-2" 1`] = `
<h2
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 krARBI"
  fontFamily="default"
  fontSize="xxl"
  spacing="base"
/>
`;

exports[`Heading should match snapshot with scale="level-3" 1`] = `
<h3
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 ijReaP"
  fontFamily="default"
  fontSize="xl"
  spacing="base"
/>
`;

exports[`Heading should match snapshot with scale="level-4" 1`] = `
<h4
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 cRKbib"
  fontFamily="default"
  fontSize="lg"
  spacing="base"
/>
`;

exports[`Heading should match snapshot with scale="level-5" 1`] = `
<h5
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 jYOSgS"
  fontFamily="default"
  fontSize="md"
  spacing="half"
/>
`;
